chain/src/chain.rs

```diff
@@ -398,8 +399,9 @@ impl ChainService {
             );
             self.find_fork(&mut fork, current_tip_header.number(), &block, ext);
             if !fork.detached_blocks.is_empty() {
-                metrics!(gauge, "ckb-chain.reorg", fork.attached_blocks.len() as i64, "type" => "attached");
-                metrics!(gauge, "ckb-chain.reorg", fork.detached_blocks.len() as i64, "type" => "detached");
+                // We care about "the size of reorg", then the length of attached blocks covers it.
+                let attached_blocks = fork.attached_blocks().len();
+                metrics!(counter, "ckb.reorg_attached_blocks", attached_blocks as u64);
```

Comment:
I think detached blocks length also a useful metric here, or it just always being `attached_blocks + 1`?

Reply:
What we actually care about are:
  - The frequency of reorg happens
  - The scale of reorg

Back to the discussion. I agree that the length of the detached is meaningful. However, most of the time, detached-length is very close to attached-length(`detached + 1 == attached`). We can estimate the scale of reorg based attached-length only, that's enough for us.